crates/nargo_cli/src/manifest.rs

```diff
@@ -1,13 +1,249 @@
+}
+
+#[derive(Default, Debug, Deserialize, Clone)]
+#[serde(rename_all = "kebab-case")]
+struct WorkspaceConfig {
+    /// List of members in this workspace.
+    members: Vec<PathBuf>,
+    /// Specifies the default crate to interact with in the context (similarly to how we have nargo as the default crate in this repository).
+    default_member: Option<PathBuf>,
+}
+
+#[allow(dead_code)]
+#[derive(Default, Debug, Deserialize, Clone)]
+struct PackageMetadata {
+    name: String,
+    description: Option<String>,
+    authors: Option<Vec<String>>,
+    // If not compiler version is supplied, the latest is used
+    // For now, we state that all packages must be compiled under the same
+    // compiler version.
+    // We also state that ACIR and the compiler will upgrade in lockstep.
+    // so you will not need to supply an ACIR and compiler version
+    compiler_version: Option<String>,
+    backend: Option<String>,
+    license: Option<String>,
+}
+
+#[derive(Debug, Deserialize, Clone)]
+#[serde(untagged)]
+/// Enum representing the different types of ways to
+/// supply a source for the dependency
+enum DependencyConfig {
+    Github { git: String, tag: String },
+    Path { path: String },
+}
+
+impl DependencyConfig {
+    fn to_dependency(&self, pkg_root: &Path) -> Result<package::Dependency, ManifestError> {
+        match self {
+            Self::Github { git, tag } => {
+                let dir_path = clone_git_repo(git, tag).map_err(ManifestError::GitError)?;
+                let (entry_path, crate_type) = super::lib_or_bin(&dir_path)?;
+                let workspace = resolve_workspace(&dir_path, None)?;
+                Ok(package::Dependency::Remote { entry_path, crate_type, workspace })
+            }
+            Self::Path { path } => {
+                let dir_path = pkg_root.join(path);
+                let (entry_path, crate_type) = super::lib_or_bin(&dir_path)?;
+                let workspace = resolve_workspace(&dir_path, None)?;
+                Ok(package::Dependency::Local { entry_path, crate_type, workspace })
+            }
+        }
+    }
+}
+
+fn toml_in_directory(dir_path: &Path) -> Result<NargoToml, ManifestError> {
+    let toml_path = super::find_package_manifest(dir_path)?;
     let toml_as_string =
-        std::fs::read_to_string(&path_to_toml).expect("ice: path given for toml file is invalid");
+        std::fs::read_to_string(toml_path).expect("ice: path given for toml file is invalid");
+    let nargo_toml =
+        NargoToml { root_dir: dir_path.to_path_buf(), config: toml_as_string.try_into()? };
+
+    Ok(nargo_toml)
+}
+
+/// Resolves a Nargo.toml file into a `Workspace` struct as defined by our `nargo` core.
+/// Recursively resolves filesystem locations and turns them into `Package` structs
+pub(crate) fn resolve_workspace(
+    dir_path: &Path,
+    selected_package: Option<CrateName>,
+) -> Result<Workspace, ManifestError> {
+    let nargo_toml = toml_in_directory(dir_path)?;
```

Comment:
TODO: Revisit taking the dir_path because we might not want to "find up" like the `toml_in_directory` function does right now